server/src/main/java/io/crate/expression/udf/UserDefinedFunctionService.java

```diff
@@ -123,7 +133,8 @@ void dropFunction(final String schema,
                       final List<DataType<?>> argumentTypes,
                       final boolean ifExists,
                       final ActionListener<AcknowledgedResponse> listener,
-                      final TimeValue timeout) {
+                      final TimeValue timeout,
+                      Schemas schemas) {
```

Comment:
We must pass in the schemas here instead of getting it as a guice singleton on the ctor to avoid circular dependencies.